### PR TITLE
oauth2 flow access code

### PR DESCRIPTION
**Pull request by bodnia**
@ponelat is this what you meant by supporting both ajax and redirect solutions?

*DO NOT MERGE*

### CODE REVIEW

File: dist/oauth2-redirect.html [hunk 1]
```diff
@@ -33,7 +33,41 @@
             if (qp.code) {
                 delete oauth2.state;
                 oauth2.auth.code = qp.code;
-                createForm(oauth2.auth, qp).submit();
+                var form = new FormData(createForm(oauth2.auth, qp));
+                var headers = new Headers();
+
+                headers.append("Access-Control-Allow-Origin", "*");
+                fetch(oauth2.auth.schema.get("tokenUrl"), {
```
Comment (ponelat) [inline dist/oauth2-redirect.html]: If you post this against echo.ponelat.com, what do you see? The same as if you were use the `<form/>` method?

File: dist/oauth2-redirect.html [hunk 2]
```diff
@@ -33,7 +33,41 @@
...
+                  method: "POST",
+                  headers: headers,
+                  mode: "cors",
+                  body: form
+                }).then(function (res) {
+                    var status = res.status
+                    if (status >= 300 && status < 400) {
+                      window.location = res.headers && res.headers.location
```
Comment (ponelat) [inline dist/oauth2-redirect.html]: I think `res.headers` is guaranteed to exist.
Is this _native fetch_ or swagger-js's `fetch`?

File: dist/oauth2-redirect.html [hunk 3]
```diff
@@ -61,7 +93,7 @@
             "grant_type": "authorization_code",
             "client_secret": auth.clientSecret,
             "client_id": auth.clientId
-        }
+        };
```
Comment (ponelat) [inline dist/oauth2-redirect.html]: watch out for lint errors :)

**Comment by ponelat** [comment]
I can't say I understand the flows here entirely, given a quick glance. But aside from that, we should be able to use fetch over the form.submit() way.

I'd  really like a document describing our oAuth flows and how _we_ expect them to work, wrt the oAuth 2.0 specification and OAS _flows_. @webron thoughts?

**Comment by webron** [comment]
Not sure what you mean, @ponelat. The OAS spec doesn't specify how the flows are to be implemented, but which (all, for that matter). The how simply needs to follow the OAauth2 spec.

**Pull request closed without merging**